modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/transcripts/TranscriptSearchViewModel.kt

```diff
@@ -0,0 +1,114 @@
+package au.com.shiftyjelly.pocketcasts.player.view.transcripts
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import au.com.shiftyjelly.pocketcasts.utils.log.LogBuffer
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.debounce
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+
+private const val SEARCH_DEBOUNCE = 300L
```

Comment:
Added a small debounce. I tried without debounce also and didn't notice any performance issues even for large text. Should we keep it or not?